extension/src/background/helpers/dataStorage.ts

```diff
@@ -290,6 +293,31 @@ export const addBlockaidAnnouncedIsEnabled = async () => {
   }
 };
 
+export const migrateAllowlistToKeyNetworkSchema = async () => {
+  const localStore = dataStorageAccess(browserLocalStorage);
+  const storageVersion = (await localStore.getItem(STORAGE_VERSION)) as string;
+
+  if (!storageVersion || semver.lt(storageVersion, "4.6.0")) {
+    const currentAllowlist = await localStore.getItem(ALLOWLIST_ID);
+
+    if (currentAllowlist) {
+      const lastUsedAccount = await localStore.getItem(LAST_USED_ACCOUNT);
```

Comment:
On migration, I didn't want to erase a user's current allowlist, but it also felt like a security risk to automatically opt all their Mainnet accounts into the existing allow list. As a compromise, I'm assigning any existing allowlist to the user's `lastUsedAccount` on Testnet